### PR TITLE
Closes issue #1201 outline for code liniting + eslint instructions 

### LINKED ISSUE #1201: Add outline for code liniting + eslint instructions
We want our participants to submit clean PR's and see what well-formatted code looks like. Once they learn syntax and formatting manually, we need a new lesson for how to set up and use a linter.

Create a new lesson outline (markdown only, no need for slides or recording) using our [TEMPLATE](https://github.com/Techtonica/curriculum/blob/master/_templates/topic-outline.md).

### Topics to Cover
- Why automated linting?
- eslint setup and use
- suggestion to use popular default style guides, e.g. AirBNB
- Integration with vscode to automatically show linting violations for JavaScript

### PR DESCRIPTION
**Pull request by guptavarun619**
Fixes #1201 Add outline for code liniting + eslint instructions

I've compiled a markdown `code-linting-eslint.md` with appropriate amount of instructions and resources for the topic. I wasn't able to figure out where to put it so i created `./linting` directory in root and here's my PR. Please have look and share your valuable feedback

### CODE REVIEW

**Comment by Kundan28** [comment]
Please link the issue to the PR.

File: linting/code-linting-eslint.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+- Lesson: 5-7 min
+- Guided Practice: 5 min (Mock Project)
+- Independed Practice: (Implement it with your projects)
+- Check for Understanding: 1 min
+
+### Prerquisites
+
+Here are topics you must be comfortable with to get the maximum benefit out of this topic:
+
+-  [JavaScript](https://github.com/Techtonica/curriculum/blob/master/javascript/javascript-1.md)
```
Comment (vegetabill) [inline linting/code-linting-eslint.md]: The prereqs should be:
- [Javascript 6 - Object Literals](https://github.com/Techtonica/curriculum/blob/master/javascript/javascript-6-object-literals.md)
- [VSCode](https://github.com/Techtonica/curriculum/blob/master/vs-code/vscode.md)

Please note all links to other curriculum materials must be relative links.

File: linting/code-linting-eslint.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+-  [ESLint Quickstart - find errors automatically](https://www.youtube.com/watch?v=qhuFviJn-es) - freeCodeCamp.org
+-  [ESLint + Prettier + VS Code — The Perfect Setup](https://www.youtube.com/watch?v=lHAeK8t94as) - Web Bos
+
+
+### Lesson
+
+ESLint is a tool for identifying and reporting on patterns found in ECMAScript/JavaScript code. In many ways, it is similar to JSLint and JSHint with a few exceptions:
+
+- ESLint uses Espree for JavaScript parsing. 
+- ESLint uses an AST to evaluate patterns in code.
```
Comment (vegetabill) [inline linting/code-linting-eslint.md]: This is irrelevant to a beginner. Please cut this or move to supplemental materials if there is interesting reading about it.

File: linting/code-linting-eslint.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+```
+$ npm install eslint --save-dev
+```
+
+- To install ESLint globally:
+
+```
+$ npm install eslint -g
+```
```
Comment (vegetabill) [inline linting/code-linting-eslint.md]: They should install it per-project I believe.

File: linting/code-linting-eslint.md [hunk 4]
```diff
@@ -0,0 +1,143 @@
+
+# Code liniting + ESLint Instructions
+
+### Projected Time
+10-15 minutes
```
Comment (vegetabill) [inline linting/code-linting-eslint.md]: Please make this 30 minutes.

File: linting/code-linting-eslint.md (hunk 1 above)
Comment (guptavarun619) [reply linting/code-linting-eslint.md]: I'll be right on it. The links provided are include:
- YouTube tutorial for setting up eslint
- Official ESLint Documentation
- Official links to ESLint extension(VSCode, Brackets, Sublime text 3 etc.)

File: linting/code-linting-eslint.md (hunk 3 above)
Comment (guptavarun619) [reply linting/code-linting-eslint.md]: I've suggested to install ESLint per-project before providing the command for both. Do you still want me to remove instruction for global-installation?

File: javascript/eslint.md [hunk 5]
```diff
@@ -0,0 +1,139 @@
+
+# Code liniting + ESLint Instructions
+
+### Projected Time
+30 minutes
+
+- Lesson: 15-20 min
+- Guided Practice: 5 -10 min (Mock Project)
```
Comment (Kundan28) [inline javascript/eslint.md]: Remove whitespace after `5`.

File: javascript/eslint.md [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+- Independed Practice: (Implement it with your projects)
+- Check for Understanding: 1 min
+
+### Prerquisites
+
+Here are topics you must be comfortable with to get the maximum benefit out of this topic:
+
+-  [Javascript 6 - Object Literals](https://github.com/Techtonica/curriculum/blob/master/javascript/javascript-6-object-literals.md)
```
Comment (Kundan28) [inline javascript/eslint.md]: Remove extra whitespace between `-` & `[Javascript...]

**Comment by guptavarun619** [comment]
I've removed extra whitespaces as pointed out by @Kundan28. @vegetabill  Please have a look at the installation section and consider my reply to the change asked by you for it.

File: linting/code-linting-eslint.md (hunk 3 above)
Comment (vegetabill) [reply linting/code-linting-eslint.md]: Yes that is what I requested.

**Comment by vegetabill** [comment]
Looks great thanks @guptavarun619 !

**Comment by vegetabill** [comment]
@Kundan28 I've confirmed this on the score sheet.

**Comment by Kundan28** [comment]
> @Kundan28 I've confirmed this on the score sheet.

Thanks !

**Pull request merged**